### ISSUE TITLE
delete method must have body

### ISSUE THREAD

**Issue by jweboy** [fastify/fastify#894]
I remember that http delete method supports sending header information to complete a request without sending the body of the request body. However, when sending a delete request with fastify, it is found that it must take the body. Otherwise, there will be json error. The error is as follows:
```js
{
    "statusCode": 400,
    "error": "Bad Request",
    "message": "Unexpected end of body input"
}
```
I would like to ask if this is a bug or design?

**Comment by mcollina** [comment]
IMHO it's a bug.

**Comment by jsumners** [comment]
Bug.

**Comment by mcollina** [comment]
Would you like to send a PR?

**Comment by jweboy** [comment]
My problem has been solved, thank you.

**Comment by mancioshell** [comment]
This problem has been solved in 1.8.0 ?

**Comment by delvedor** [comment]
@mancioshell yes, but we are updating the solution in https://github.com/fastify/fastify/pull/1027.
